src/langchain_google_firestore/vectorstores.py

```diff
@@ -0,0 +1,405 @@
+
+from .version import __version__
+
+USER_AGENT = "langchain-google-firestore-python:vectorstore" + __version__
+IMPORT_ERROR_MSG = """`google-cloud-firestore` package not found,
+please run `pip3 install google-cloud-firestore`"""
+WRITE_BATCH_SIZE = 500
+DEFAULT_TOP_K = 4
+
+
+class FirestoreVectorStore(VectorStore):
+    """Interface for vector store."""
+
+    def __init__(
+        self,
+        collection: CollectionReference | str,
+        embedding: Embeddings,
+        client: Optional[Client] = None,
+        content_field: str = "content",
+        metadata_field: str = "metadata",
+        embedding_field: str = "embedding",
+        distance_strategy: Optional[DistanceMeasure] = DistanceMeasure.COSINE,
+        filters: Optional[BaseFilter] = None,
+    ) -> None:
+        """Constructor for FirestoreVectorStore.
+
+        Args:
+            source: The source collection or document reference to store the data.
+            embedding: The embeddings to use for the vector store.
+            client: The Firestore client to use. If not provided, a new client will be created.
+            content_field: The field name to store the content data.
+            metadata_field: The field name to store the metadata.
+            embedding_field: The field name to store the text embeddings.
+            distance_strategy: The distance strategy to use for calculating distances
+            between vectors. Defaults to DistanceStrategy.COSINE.
+
+        Raises:
+            ImportError: If the `firestore` package is not found.
+        """
+        try:
+            from google.cloud import firestore  # type: ignore
+        except ModuleNotFoundError as exc:
+            raise ImportError(IMPORT_ERROR_MSG) from exc
+
+        # Check if the client is provided, otherwise create a new client with
+        # the default client info.
+        self.client = client or firestore.Client()
+
+        client_agent = self.client._client_info.user_agent
+        if not client_agent:
+            self.client._client_info.user_agent = USER_AGENT
+        elif USER_AGENT not in client_agent:
+            self.client._client_info.user_agent = " ".join([client_agent, USER_AGENT])
+
+        if isinstance(collection, str):
+            self.collection = self.client.collection(collection)
+        else:
+            self.collection = collection
+
+        self.embedding = embedding
+        self.content_field = content_field
+        self.metadata_field = metadata_field
+        self.embedding_field = embedding_field
+        self.distance_strategy = distance_strategy
+        self.filters = filters
+
+    @property
+    def embeddings(self) -> Optional[Embeddings]:
+        return self.embedding
+
+    def add_texts(
+        self,
+        texts: Iterable[str],
+        metadatas: Optional[List[dict]] = None,
+        ids: Optional[List[str]] = None,
+        **kwargs: Any,
```

Comment:
Is the kwargs unused?

-- I don't see where we use it. If this is the case, can we remove it?

Reply:
mypy will complain:
```
Signature of "delete" incompatible with supertype "VectorStore"Mypy[override](https://mypy.readthedocs.io/en/latest/_refs.html#code-override)
```